update to v5

kirsty-hames opened the pull request:
- hbs/less refactor
- separate plugin/theme less
- bower/readme bump
- js function name update
- tabs to spacing

kirsty-hames committed:
initial v5 commit

- hbs/less refactor
- separate plugin/theme less
- bower/readme bump
- js function name update
- tabs to spacing

kineojen reviewed (changes requested):
I'm showing tabs instead of spaces in adapt-hint.js and hint.hbs.

moloko commented:
I'm going to do a bit of a tidy up of the code so leave this with me

moloko committed:
better component description

moloko committed:
add support for displaying hint in Notify popup (for optimum accessibility)
change 'open' event name to from 'js-hint-widget:open' to 'hint:opened' to be more in line with adapt standard event naming
refactor js code to simplify some of it, remove duplication
Switch over to new notify/a11y APIs

moloko commented:
@kirsty-hames I copied this from B**ys templates, could you just check I haven't copied anything into the .less that I shouldn't have? ta ;-)

otherwise I think this is ready for review... the a11y support for the non-notify version is not as good as it could be, but then we do recommend use of the Notify popup for best a11y support so IMO it's 'good enough'.

guywillis committed:
Converted tabs to spaces, indentation amends

guywillis committed:
Less rework

tomgreenfield commented on js/adapt-hint.js:
```diff
@@ -1,120 +1,102 @@
+  'core/js/adapt'
+], function (Adapt) {
+
+  var HintPopupView = Backbone.View.extend({
```
Might as well also convert the `var` here, and on line 20, to `const`.

guywillis replied:
Done

tomgreenfield commented on js/adapt-hint.js:
```diff
@@ -1,120 +1,102 @@
+      const globals = Adapt.course.get('_globals')._extensions._hint;
+      const isBeingOpened = $widget.hasClass('is-closed');
+
+      $widget.toggleClass('is-open', isBeingOpened).toggleClass('is-closed', !isBeingOpened);
+      $button.attr('aria-label', (isBeingOpened ? globals.closeButtonText : globals.openButtonText))
+        .toggleClass('is-open', isBeingOpened).toggleClass('is-closed', !isBeingOpened);
```
As the modification classes are now identical, can the Less be reworked to use them from a shared parent higher up in the DOM, @guywillis?

For example, with the below or similar:
```js
this.$el
  .toggleClass('is-open', isBeingOpened)
  .toggleClass('is-closed', !isBeingOpened);
```

guywillis replied:
Whilst it would make for more optimised code, making this change would make the HTML and Less slightly less easier to read in my opinion. 

I've left the current implementation but made the code a little easier to read.

guywillis committed:
Changed var to const, legibility update

kirsty-hames commented:
I can't approve as I opened the PR but it's a +1 from me thanks

guywillis committed:
Cleaner offset

guywillis commented:
One final commit to make the offset cleaner, sorry!

moloko commented:
everybody happy with this now?
